libs/expo/betterangels/src/lib/screens/AddEditClient/index.tsx

```diff
@@ -62,24 +64,90 @@ export default function AddEditClient({ id }: { id?: string }) {
       },
     ],
   });
-  const [updateClient] = useUpdateClientProfileMutation();
-  const [createClient] = useCreateClientProfileMutation({
-    refetchQueries: [
-      {
-        query: ClientProfilesDocument,
-        variables: {
-          pagination: { limit: 20 + 1, offset: 0 },
-          filters: {
-            search: '',
-          },
-          order: {
-            user_FirstName: Ordering.AscNullsFirst,
-            id: Ordering.Desc,
+  const [updateClient, { error: updateMutationError }] =
+    useUpdateClientProfileMutation();
+
+  const [createClient, { error: createMutationError }] =
+    useCreateClientProfileMutation({
+      refetchQueries: [
+        {
+          query: ClientProfilesDocument,
+          variables: {
+            pagination: { limit: 20 + 1, offset: 0 },
+            filters: {
+              search: '',
+            },
+            order: {
+              user_FirstName: Ordering.AscNullsFirst,
+              id: Ordering.Desc,
+            },
           },
         },
-      },
-    ],
-  });
+      ],
+    });
+
+  const extractErrorFields = (errors: ApolloError) => {
+    return errors.graphQLErrors
+      .map((error: GraphQLFormattedError) => {
+        const regex = /data\.(.*?);/;
+        const match = error.message.match(regex);
+        return match ? match[1] : null;
+      })
+      .filter(Boolean);
+  };
```

Comment:
This extracts out the node that's after `data` from the error message, and sets the error.

E.g., `[ApolloError: Variable '$data' got invalid value '1231231313' at 'data.phoneNumber'; ]` - extracts `phoneNumber`

E.g., [ApolloError: Variable '$data' got invalid value '1231231313' at 'data.contact[0].phoneNumber'; ] - extracts `contact[0].phoneNumber`

Reply:
I'm  thinking why not require a user to write correct number on front end but send it to back, check it, throw error, show error. 

just validate with react-hook form and show front error message. 

lets walk through this together